packages/client-lib/src/store/mocks.ts

```diff
@@ -0,0 +1,22 @@
+import { Store } from "./types";
+
+export function makeMockStore<State extends object>(initialState: State) {
+  const setPropertyMock = jest.fn();
+  const resetMock = jest.fn();
+
+  const mockStore: Store<State> = {
+    state: initialState,
+    setProperty: setPropertyMock,
+    reset: resetMock,
+  };
```

Comment:
Can't use `jest.Mocked` here because that expect everything in the object to be mocked. Can't really mock the plain object `state`...?